File: backend/app/schemas/daily_bounty.py
"""Daily Bounty schemas."""

from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field


class DailyBountyProblem(BaseModel):
    """일일 현상금 문제 정보"""
    id: int
    slug: str
    title: str
    difficulty: str
    domain: str
    short_description: Optional[str] = None


class DailyBountyRewards(BaseModel):
    """보상 정보"""
    base_reward: int = Field(5, description="기본 보상 토큰")
    streak_bonus: int = Field(50, description="연속 보너스 토큰")
    streak_threshold: int = Field(3, description="보너스 지급 연속 일수")


class DailyBountyCompletion(BaseModel):
    """완료 상태"""
    base_reward_granted: bool = False
    streak_bonus_granted: bool = False


class DailyBountyStatusResponse(BaseModel):
    """일일 현상금 상태 응답"""
    date: str = Field(..., description="날짜 (YYYY-MM-DD)")
    problem: Optional[DailyBountyProblem] = Field(None, description="오늘의 문제")
    is_completed: bool = Field(False, description="오늘 완료 여부")
    user_streak: int = Field(0, description="현재 연속 일수")
    streak_progress: int = Field(0, description="보너스까지 남은 일수 (0-2)")
    time_remaining_seconds: int = Field(0, description="자정까지 남은 초")
    rewards: DailyBountyRewards = Field(default_factory=DailyBountyRewards)
    completion: Optional[DailyBountyCompletion] = None
    error: Optional[str] = None

    class Config:
        from_attributes = True


class DailyBountyHistoryItem(BaseModel):
    """완료 기록 아이템"""
    date: str
    problem_id: Optional[int]
    streak_count: int
    base_reward_granted: bool
    streak_bonus_granted: bool


class DailyBountyHistoryResponse(BaseModel):
    """완료 기록 응답"""
    completions: List[DailyBountyHistoryItem]
    current_streak: int
    total_completions: int

File: frontend/types/daily-bounty.ts
/**
 * Daily Bounty (일일 현상금) Types
 */

export interface DailyBountyProblem {
  id: number;
  slug: string;
  title: string;
  difficulty: string;
  domain: string;
  short_description?: string;
}

export interface DailyBountyRewards {
  base_reward: number;      // 기본 보상 (5)
  streak_bonus: number;     // 스트릭 보너스 (50)
  streak_threshold: number; // 보너스 조건 (3일)
}

export interface DailyBountyCompletion {
  base_reward_granted: boolean;
  streak_bonus_granted: boolean;
}

export interface DailyBountyStatus {
  date: string;                          // YYYY-MM-DD
  problem: DailyBountyProblem | null;
  is_completed: boolean;
  user_streak: number;                   // 현재 연속 일수
  streak_progress: number;               // 3일까지 남은 일수 (0-2)
  time_remaining_seconds: number;        // 자정까지 남은 시간
  rewards: DailyBountyRewards;
  completion?: DailyBountyCompletion;
  error?: string;
}

export interface DailyBountyHistoryItem {
  date: string;
  problem_id: number | null;
  streak_count: number;
  base_reward_granted: boolean;
  streak_bonus_granted: boolean;
}

export interface DailyBountyHistory {
  completions: DailyBountyHistoryItem[];
  current_streak: number;
  total_completions: number;
}
